app/src/main/java/org/kinecosystem/kinit/viewmodel/SupportViewModel.kt

```diff
@@ -0,0 +1,154 @@
+package org.kinecosystem.kinit.viewmodel
+
+import android.databinding.ObservableBoolean
+import android.databinding.ObservableField
+import android.util.Log
+import android.webkit.JavascriptInterface
+import org.kinecosystem.kinit.KinitApplication
+import org.kinecosystem.kinit.analytics.Analytics
+import org.kinecosystem.kinit.analytics.Events
+import org.kinecosystem.kinit.repository.UserRepository
+import org.kinecosystem.kinit.util.DeviceUtils
+import org.kinecosystem.kinit.util.GeneralUtils
+import java.lang.Exception
+import javax.inject.Inject
+
+class SupportViewModel(val destination: Destination, urlParams: Map<String, String?>?) {
+
+    @Inject
+    lateinit var userRepository: UserRepository
+
+    enum class Destination {
+        FAQ,
+        CONTACT_US,
+        FEEDBACK
+
+    }
+
+    class FAQAnalyticsData(json: String){
+        private val jsonObj = GeneralUtils.stringToJson(json)
+        var faqCategory: String? = null
+        var faqSubCategory: String? = null
+        var isHelpful: Boolean? = null
+        var formData: String? = null
+        var serverError: String? = null
+
+        init {
+            faqCategory = get("faqCategory")
+            faqSubCategory = get("faqSubCategory")
+            isHelpful = get("isHelpful")
+            formData = get("formData")
+            serverError = get("serverError")
+        }
+
+        fun <T: Any> get(name: String): T?{
+            try {
+                return jsonObj?.get(name) as T?
+            } catch (e: Exception){
+                Log.i("FAQAnalyticsData", "$name was not found")
+            }
+            return null
```

Comment:
this may return null but if it does for one of the fields that WAS expected, we will crash when trying to send the event. 
Need to either return empty string, or "not available" or check for null when using this